trace/google/cloud/trace/span_context.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""SpanContext encapsulates the current context within the request's trace."""
+
+from google.cloud.trace.trace import generate_trace_id
+
+import re
+import logging
+
+_TRACE_CONTEXT_HEADER_FORMAT = '([0-9a-f]{32})(\/(\d+))?(;o=(\d+))?'
+_TRACE_HEADER_KEY = 'X_CLOUD_TRACE_CONTEXT'
+_TRACE_ID_FORMAT = '[0-9a-f]{32}?'
+
+
+class SpanContext(object):
+    """SpanContext includes 3 fields: traceId, spanId, and an enabled flag 
+    which indicates whether or not the request is being traced. It contains the
+    current context to be propagate to the child spans.
+    
+    :type trace_id: str
+    :param trace_id: (Optional) Trace_id is a 32 digits uuid for the trace.
+                     If not given, will generate one automatically.
+
+    :type span_id: int
+    :param span_id: (Optional) Identifier for the span, unique within a trace.
+
+    :type enabled: bool
+    :param enabled: (Optional) Indicates whether the request is traced or not.
+    
+    :type from_header: bool
+    :param from_header: (Optional) Indicates whether the trace context is
+                        generated from request header.
+    """
+    def __init__(
+            self,
+            trace_id=None,
+            span_id=None,
+            enabled=True,
+            from_header=False):
+        if trace_id is None:
+            trace_id = generate_trace_id()
+
+        self.trace_id = self.check_trace_id(trace_id)
+        self.span_id = self.check_span_id(span_id)
+        self.enabled = enabled
+        self.from_header = from_header
+
+    def check_span_id(self, span_id):
```

Comment:
Thanks for this information! I just have a question that what should I return if the span_id or trace_id is not valid? Currently I generate a new one instead to create the SpanContext and output a warning log (which means ignore the current context), but it might also make sense to throw exception and stop creating SpanContext. Which way do you prefer?